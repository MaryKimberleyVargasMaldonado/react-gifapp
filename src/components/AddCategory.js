import React, { useState } from "react";
import { ReactDOM } from "react";
import PropTypes from "prop-types";

export const AddCategory = (props) => {
	const [inputValue, setInputValue] = useState("Search a category:");

	// H A N D L E   I N P U T   C H A N G E
	const handleInputChange = (e) => {
		setInputValue(e.target.value);
		console.log("HandleInputChange llamado.");
	};

	//colocar el event a un form
	// H A N D L E   S U B M I T
	const handleSubmit = (e) => {
		e.preventDefault(); //prevenir comportamiento por defecto del formulario.
		console.log("Submit done", inputValue);

		if (inputValue.trim().length > 2) {
			props.setCategories((cate) => [inputValue, ...cate]);
			setInputValue("");
		}
	};

	return (
		//FRAGMENT
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<h2>{inputValue}</h2>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

// P R O P E R T I E S
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
