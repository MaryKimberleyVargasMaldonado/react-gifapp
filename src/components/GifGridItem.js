import React from "react";
//import { GifGrid } from "./components/GiftGrid";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
	console.log({ title, url });
	return (
		<div className="card animate__animated animate__zoomInDown">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

// P R O P E R T I E S   -   P R O P T Y P E S
GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
	//id: PropTypes.string.isRequired,
};
