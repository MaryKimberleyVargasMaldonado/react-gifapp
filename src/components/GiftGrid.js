import React from "react";
import { useState, useEffect, Fragment } from "react/cjs/react.development";
//import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
	// U S E   F E T C H   G I F T S
	const { data: images, loading } = useFetchGifs(category);

	// I M A G E S
	/*
	const [images, setImages] = useState([]);


*/
	//getGifs();
	return (
		<Fragment>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			{loading && <p className="animate__animate animate__flash">Loading...</p>}
			<div className="card-grid">
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</Fragment>
	);
};

/*
			<ol>
				{images.map((img) => (
					<li key={img.id}>{img.title}</li>
				))}
			</ol>*/
