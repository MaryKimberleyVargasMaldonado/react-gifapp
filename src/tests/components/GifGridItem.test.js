import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";
import React from "react";
import "@testing-library/jest-dom";

// Inicio del test
describe("Prueba en GifGridItem.", () => {
	/*	beforeEach(() => {
		wrapper = shallow(<GifGridItem />);
	});*/

	const title = "Un titulo";
	const url = "https://localhost/algo.jpg";

	const wrapper = shallow(<GifGridItem url={url} title={title} />);

	// P R I M E R   T E S T
	test("1. Title y URL obligatorios: ", () => {
		expect(wrapper).toMatchSnapshot();
	}); // p

	//  S E G U N D O  T E S T
	test("2. Title debe tener un pÃ¡rrafo: ", () => {
		const p = wrapper.find("p").text().trim();
		//expect(p.text().trim()).toBe(title);
		expect(p).toBe(title);
	}); // s

	// T E R C E R   T E S T
	test("3. Debe tener una imagen igual al props: ", () => {
		const img = wrapper.find("img");
		expect(img.prop("src")).toBe(url);
		expect(img.prop("alt")).toBe(title);
	}); // tt
}); // describe
