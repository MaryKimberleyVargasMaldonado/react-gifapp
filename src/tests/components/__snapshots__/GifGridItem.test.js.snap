// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGridItem. 1. Title y URL obligatorios:  1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
